## Read and Clean Dataset


Import the libraries.


In [1]:
# Install tensorflow and keras in Anaconda
# !pip install tensorflow
# !pip install keras

In [2]:
import pandas as pd
import numpy as np
import keras
from keras.models import Sequential
from keras.layers import Dense
from sklearn.model_selection import train_test_split
from sklearn import metrics

<strong>The dataset is about the compressive strength of different samples of concrete based on the volumes of the different ingredients that were used to make them. Ingredients include:</strong>

<strong>1. Cement</strong>

<strong>2. Blast Furnace Slag</strong>

<strong>3. Fly Ash</strong>

<strong>4. Water</strong>

<strong>5. Superplasticizer</strong>

<strong>6. Coarse Aggregate</strong>

<strong>7. Fine Aggregate</strong>


Read the data into a <em>pandas</em> dataframe.


In [3]:
concrete_data = pd.read_csv('concrete_data.csv')
concrete_data.head()

Cement  Blast Furnace Slag  Fly Ash  Water  Superplasticizer  \
0   540.0                 0.0      0.0  162.0               2.5   
1   540.0                 0.0      0.0  162.0               2.5   
2   332.5               142.5      0.0  228.0               0.0   
3   332.5               142.5      0.0  228.0               0.0   
4   198.6               132.4      0.0  192.0               0.0   

   Coarse Aggregate  Fine Aggregate  Age  Strength  
0            1040.0           676.0   28     79.99  
1            1055.0           676.0   28     61.89  
2             932.0           594.0  270     40.27  
3             932.0           594.0  365     41.05  
4             978.4           825.5  360     44.30

So the first concrete sample has 540 cubic meter of cement, 0 cubic meter of blast furnace slag, 0 cubic meter of fly ash, 162 cubic meter of water, 2.5 cubic meter of superplaticizer, 1040 cubic meter of coarse aggregate, 676 cubic meter of fine aggregate. Such a concrete mix which is 28 days old, has a compressive strength of 79.99 MPa.


#### Let's check how many data points we have.


In [4]:
concrete_data.shape

(1030, 9)

So, there are approximately 1000 samples to train our model on. Because of the few samples, we have to be careful not to overfit the training data.


Let's check the dataset for any missing values.


In [5]:
concrete_data.describe()

Cement  Blast Furnace Slag      Fly Ash        Water  \
count  1030.000000         1030.000000  1030.000000  1030.000000   
mean    281.167864           73.895825    54.188350   181.567282   
std     104.506364           86.279342    63.997004    21.354219   
min     102.000000            0.000000     0.000000   121.800000   
25%     192.375000            0.000000     0.000000   164.900000   
50%     272.900000           22.000000     0.000000   185.000000   
75%     350.000000          142.950000   118.300000   192.000000   
max     540.000000          359.400000   200.100000   247.000000   

       Superplasticizer  Coarse Aggregate  Fine Aggregate          Age  \
count       1030.000000       1030.000000     1030.000000  1030.000000   
mean           6.204660        972.918932      773.580485    45.662136   
std            5.973841         77.753954       80.175980    63.169912   
min            0.000000        801.000000      594.000000     1.000000   
25%            0.000000        932.000000      730.950000     7.000000   
50%            6.400000        968.000000      779.500000    28.000000   
75%           10.200000       1029.400000      824.000000    56.000000   
max           32.200000       1145.000000      992.600000   365.000000   

          Strength  
count  1030.000000  
mean     35.817961  
std      16.705742  
min       2.330000  
25%      23.710000  
50%      34.445000  
75%      46.135000  
max      82.600000

In [6]:
concrete_data.isnull().sum()

Cement                0
Blast Furnace Slag    0
Fly Ash               0
Water                 0
Superplasticizer      0
Coarse Aggregate      0
Fine Aggregate        0
Age                   0
Strength              0
dtype: int64

The data looks very clean and is ready to be used to build our model.


#### Split data into predictors (X) and target (y)


The target variable in this problem is the concrete sample strength. Therefore, our predictors will be all the other columns.


In [7]:
concrete_data_columns = concrete_data.columns

X = concrete_data[concrete_data_columns[concrete_data_columns != 'Strength']] # all columns except Strength
y = concrete_data['Strength'] # Strength column

<a id="item2"></a>


Let's do a quick sanity check of the predictors and the target dataframes.


In [8]:
X.head()

Cement  Blast Furnace Slag  Fly Ash  Water  Superplasticizer  \
0   540.0                 0.0      0.0  162.0               2.5   
1   540.0                 0.0      0.0  162.0               2.5   
2   332.5               142.5      0.0  228.0               0.0   
3   332.5               142.5      0.0  228.0               0.0   
4   198.6               132.4      0.0  192.0               0.0   

   Coarse Aggregate  Fine Aggregate  Age  
0            1040.0           676.0   28  
1            1055.0           676.0   28  
2             932.0           594.0  270  
3             932.0           594.0  365  
4             978.4           825.5  360

In [9]:
y.head()

0    79.99
1    61.89
2    40.27
3    41.05
4    44.30
Name: Strength, dtype: float64

### Finally, the last step is to normalize the data by substracting the mean and dividing by the standard deviation.

In [10]:
X_norm = (X - X.mean()) / X.std()
X_norm.head()

Cement  Blast Furnace Slag   Fly Ash     Water  Superplasticizer  \
0  2.476712           -0.856472 -0.846733 -0.916319         -0.620147   
1  2.476712           -0.856472 -0.846733 -0.916319         -0.620147   
2  0.491187            0.795140 -0.846733  2.174405         -1.038638   
3  0.491187            0.795140 -0.846733  2.174405         -1.038638   
4 -0.790075            0.678079 -0.846733  0.488555         -1.038638   

   Coarse Aggregate  Fine Aggregate       Age  
0          0.862735       -1.217079 -0.279597  
1          1.055651       -1.217079 -0.279597  
2         -0.526262       -2.239829  3.551340  
3         -0.526262       -2.239829  5.055221  
4          0.070492        0.647569  4.976069

Let's save the number of predictors to *n_cols* since we will need this number when building our network.


In [11]:
n_cols = X_norm.shape[1] # number of predictors
print("Number of features:", n_cols)

Number of features: 8


<a id="item1"></a>


<a id='item32'></a>


## Build a Neural Network


Let's define a function that defines our neural network model for us so that we can conveniently call it to create our model.


In [12]:
# define regression model
def create_neural_network():
    # create model
    model = Sequential()
    model.add(Dense(10, activation='relu', input_shape=(n_cols,)))
    model.add(Dense(1))
    
    # compile model
    model.compile(optimizer='adam', loss='mean_squared_error')
    return model

The above function create a model that has one hidden layers, each of 10 hidden units.


<a id="item4"></a>


<a id='item34'></a>


## Train and Test the Network


Let's call the function now to create our model.


In [13]:
# build the model
model = create_neural_network()

Next, we will train and test the model at the same time using the *fit* method. We will leave out 30% of the data for validation and we will train the model for 100 epochs.


In [14]:
# fit the model
mean_sq_error = [] # Empty list of mean squared error

def split_and_fit_model(myModel):
    X_train, X_test, y_train, y_test = train_test_split( X_norm, y, test_size=0.3, random_state=4)
    
    myModel.fit(X_train, y_train, validation_split=0.3, epochs=100, verbose=2)
    
    # evaluate the model
    scores = myModel.evaluate(X_test, y_test, verbose=0)
    print("Accuracy: {} \n Error: {}".format(scores, 100-scores*100))
    
    # Get the predicted values
    yhat = myModel.predict(X_test)
    mse_sk = metrics.mean_squared_error(y_test, yhat) # Calculate RMSE
    mean_sq_error.append(mse_sk)

# Step 4
for i in range(50):
    print("Iteration %s:\n" %str(i))
    split_and_fit_model(model)
    
print(len(mean_sq_error))
print("Mean:", np.mean(np.array(mean_sq_error)))
print("Standard Deviation :", np.std(np.array(mean_sq_error)))

Iteration 0:

Epoch 1/100
16/16 - 1s - loss: 1603.8373 - val_loss: 1488.3717 - 571ms/epoch - 36ms/step
Epoch 2/100
16/16 - 0s - loss: 1591.8435 - val_loss: 1477.0277 - 37ms/epoch - 2ms/step
Epoch 3/100
16/16 - 0s - loss: 1580.1207 - val_loss: 1465.6403 - 37ms/epoch - 2ms/step
Epoch 4/100
16/16 - 0s - loss: 1568.2263 - val_loss: 1454.3510 - 39ms/epoch - 2ms/step
Epoch 5/100
16/16 - 0s - loss: 1556.3639 - val_loss: 1442.8470 - 39ms/epoch - 2ms/step
Epoch 6/100
16/16 - 0s - loss: 1544.6816 - val_loss: 1430.8751 - 39ms/epoch - 2ms/step
Epoch 7/100
16/16 - 0s - loss: 1532.2531 - val_loss: 1419.2003 - 35ms/epoch - 2ms/step
Epoch 8/100
16/16 - 0s - loss: 1520.1205 - val_loss: 1406.9979 - 37ms/epoch - 2ms/step
Epoch 9/100
16/16 - 0s - loss: 1507.5381 - val_loss: 1394.7136 - 44ms/epoch - 3ms/step
Epoch 10/100
16/16 - 0s - loss: 1494.8850 - val_loss: 1381.8912 - 41ms/epoch - 3ms/step
Epoch 11/100
16/16 - 0s - loss: 1481.5515 - val_loss: 1369.1848 - 37ms/epoch - 2ms/step
Epoch 12/100
16/16 - 0s -

Epoch 96/100
16/16 - 0s - loss: 247.6875 - val_loss: 220.5049 - 39ms/epoch - 2ms/step
Epoch 97/100
16/16 - 0s - loss: 243.7689 - val_loss: 217.3269 - 36ms/epoch - 2ms/step
Epoch 98/100
16/16 - 0s - loss: 240.0397 - val_loss: 214.4170 - 69ms/epoch - 4ms/step
Epoch 99/100
16/16 - 0s - loss: 236.5126 - val_loss: 211.5191 - 54ms/epoch - 3ms/step
Epoch 100/100
16/16 - 0s - loss: 233.1569 - val_loss: 208.8367 - 42ms/epoch - 3ms/step
Accuracy: 237.6024932861328 
 Error: -23660.24932861328
10/10 [==============================] - 0s 1ms/step
Iteration 1:

Epoch 1/100
16/16 - 0s - loss: 229.8588 - val_loss: 206.3179 - 71ms/epoch - 4ms/step
Epoch 2/100
16/16 - 0s - loss: 226.7818 - val_loss: 203.9727 - 40ms/epoch - 2ms/step
Epoch 3/100
16/16 - 0s - loss: 223.9403 - val_loss: 201.7090 - 42ms/epoch - 3ms/step
Epoch 4/100
16/16 - 0s - loss: 221.1924 - val_loss: 199.6143 - 38ms/epoch - 2ms/step
Epoch 5/100
16/16 - 0s - loss: 218.5190 - val_loss: 197.5900 - 38ms/epoch - 2ms/step
Epoch 6/100
16/16 - 0

Epoch 90/100
16/16 - 0s - loss: 120.5547 - val_loss: 124.9842 - 37ms/epoch - 2ms/step
Epoch 91/100
16/16 - 0s - loss: 119.8618 - val_loss: 124.3934 - 36ms/epoch - 2ms/step
Epoch 92/100
16/16 - 0s - loss: 119.2424 - val_loss: 123.8733 - 34ms/epoch - 2ms/step
Epoch 93/100
16/16 - 0s - loss: 118.6006 - val_loss: 123.3692 - 35ms/epoch - 2ms/step
Epoch 94/100
16/16 - 0s - loss: 117.9862 - val_loss: 122.8203 - 39ms/epoch - 2ms/step
Epoch 95/100
16/16 - 0s - loss: 117.3460 - val_loss: 122.2828 - 37ms/epoch - 2ms/step
Epoch 96/100
16/16 - 0s - loss: 116.7643 - val_loss: 121.7004 - 38ms/epoch - 2ms/step
Epoch 97/100
16/16 - 0s - loss: 116.1990 - val_loss: 121.1616 - 41ms/epoch - 3ms/step
Epoch 98/100
16/16 - 0s - loss: 115.5545 - val_loss: 120.6458 - 44ms/epoch - 3ms/step
Epoch 99/100
16/16 - 0s - loss: 114.9933 - val_loss: 120.1009 - 43ms/epoch - 3ms/step
Epoch 100/100
16/16 - 0s - loss: 114.4551 - val_loss: 119.6102 - 43ms/epoch - 3ms/step
Accuracy: 131.6879119873047 
 Error: -13068.791198730

Epoch 85/100
16/16 - 0s - loss: 85.0698 - val_loss: 94.9640 - 37ms/epoch - 2ms/step
Epoch 86/100
16/16 - 0s - loss: 84.8839 - val_loss: 94.8384 - 36ms/epoch - 2ms/step
Epoch 87/100
16/16 - 0s - loss: 84.7267 - val_loss: 94.6806 - 36ms/epoch - 2ms/step
Epoch 88/100
16/16 - 0s - loss: 84.5368 - val_loss: 94.5567 - 36ms/epoch - 2ms/step
Epoch 89/100
16/16 - 0s - loss: 84.3993 - val_loss: 94.3934 - 36ms/epoch - 2ms/step
Epoch 90/100
16/16 - 0s - loss: 84.1846 - val_loss: 94.2514 - 37ms/epoch - 2ms/step
Epoch 91/100
16/16 - 0s - loss: 84.0179 - val_loss: 94.2325 - 50ms/epoch - 3ms/step
Epoch 92/100
16/16 - 0s - loss: 83.8383 - val_loss: 94.1219 - 38ms/epoch - 2ms/step
Epoch 93/100
16/16 - 0s - loss: 83.6617 - val_loss: 93.9848 - 69ms/epoch - 4ms/step
Epoch 94/100
16/16 - 0s - loss: 83.4987 - val_loss: 93.9009 - 38ms/epoch - 2ms/step
Epoch 95/100
16/16 - 0s - loss: 83.4245 - val_loss: 93.8168 - 39ms/epoch - 2ms/step
Epoch 96/100
16/16 - 0s - loss: 83.1901 - val_loss: 93.6814 - 36ms/epoch - 2

16/16 - 0s - loss: 68.8386 - val_loss: 81.7894 - 34ms/epoch - 2ms/step
Epoch 82/100
16/16 - 0s - loss: 68.6328 - val_loss: 81.5647 - 36ms/epoch - 2ms/step
Epoch 83/100
16/16 - 0s - loss: 68.4868 - val_loss: 81.3666 - 35ms/epoch - 2ms/step
Epoch 84/100
16/16 - 0s - loss: 68.3597 - val_loss: 81.2151 - 35ms/epoch - 2ms/step
Epoch 85/100
16/16 - 0s - loss: 68.2222 - val_loss: 81.0063 - 32ms/epoch - 2ms/step
Epoch 86/100
16/16 - 0s - loss: 68.0895 - val_loss: 80.8763 - 36ms/epoch - 2ms/step
Epoch 87/100
16/16 - 0s - loss: 67.9042 - val_loss: 80.6770 - 35ms/epoch - 2ms/step
Epoch 88/100
16/16 - 0s - loss: 67.7525 - val_loss: 80.5377 - 36ms/epoch - 2ms/step
Epoch 89/100
16/16 - 0s - loss: 67.5220 - val_loss: 80.3539 - 36ms/epoch - 2ms/step
Epoch 90/100
16/16 - 0s - loss: 67.3840 - val_loss: 80.1339 - 37ms/epoch - 2ms/step
Epoch 91/100
16/16 - 0s - loss: 67.1878 - val_loss: 79.9103 - 36ms/epoch - 2ms/step
Epoch 92/100
16/16 - 0s - loss: 67.0180 - val_loss: 79.8504 - 37ms/epoch - 2ms/step
Epoch

Epoch 78/100
16/16 - 0s - loss: 53.3162 - val_loss: 67.3607 - 40ms/epoch - 2ms/step
Epoch 79/100
16/16 - 0s - loss: 53.2092 - val_loss: 67.2260 - 39ms/epoch - 2ms/step
Epoch 80/100
16/16 - 0s - loss: 53.0900 - val_loss: 67.2969 - 39ms/epoch - 2ms/step
Epoch 81/100
16/16 - 0s - loss: 52.9562 - val_loss: 67.0957 - 40ms/epoch - 2ms/step
Epoch 82/100
16/16 - 0s - loss: 52.8818 - val_loss: 67.0499 - 34ms/epoch - 2ms/step
Epoch 83/100
16/16 - 0s - loss: 52.7395 - val_loss: 66.9203 - 36ms/epoch - 2ms/step
Epoch 84/100
16/16 - 0s - loss: 52.6465 - val_loss: 66.9306 - 34ms/epoch - 2ms/step
Epoch 85/100
16/16 - 0s - loss: 52.5286 - val_loss: 66.6700 - 35ms/epoch - 2ms/step
Epoch 86/100
16/16 - 0s - loss: 52.4539 - val_loss: 66.5103 - 34ms/epoch - 2ms/step
Epoch 87/100
16/16 - 0s - loss: 52.3630 - val_loss: 66.5388 - 36ms/epoch - 2ms/step
Epoch 88/100
16/16 - 0s - loss: 52.1538 - val_loss: 66.2527 - 35ms/epoch - 2ms/step
Epoch 89/100
16/16 - 0s - loss: 52.0278 - val_loss: 66.1429 - 37ms/epoch - 2

16/16 - 0s - loss: 44.5614 - val_loss: 57.5211 - 40ms/epoch - 2ms/step
Epoch 75/100
16/16 - 0s - loss: 44.5841 - val_loss: 57.5765 - 38ms/epoch - 2ms/step
Epoch 76/100
16/16 - 0s - loss: 44.4427 - val_loss: 57.3045 - 38ms/epoch - 2ms/step
Epoch 77/100
16/16 - 0s - loss: 44.3662 - val_loss: 57.3864 - 37ms/epoch - 2ms/step
Epoch 78/100
16/16 - 0s - loss: 44.2656 - val_loss: 57.2567 - 38ms/epoch - 2ms/step
Epoch 79/100
16/16 - 0s - loss: 44.1959 - val_loss: 57.2044 - 37ms/epoch - 2ms/step
Epoch 80/100
16/16 - 0s - loss: 44.1419 - val_loss: 57.0292 - 33ms/epoch - 2ms/step
Epoch 81/100
16/16 - 0s - loss: 44.0541 - val_loss: 57.0336 - 35ms/epoch - 2ms/step
Epoch 82/100
16/16 - 0s - loss: 43.9601 - val_loss: 56.9670 - 36ms/epoch - 2ms/step
Epoch 83/100
16/16 - 0s - loss: 43.8941 - val_loss: 56.6962 - 33ms/epoch - 2ms/step
Epoch 84/100
16/16 - 0s - loss: 43.8504 - val_loss: 56.5444 - 36ms/epoch - 2ms/step
Epoch 85/100
16/16 - 0s - loss: 43.7350 - val_loss: 56.5705 - 40ms/epoch - 2ms/step
Epoch

Epoch 71/100
16/16 - 0s - loss: 38.9991 - val_loss: 50.4460 - 39ms/epoch - 2ms/step
Epoch 72/100
16/16 - 0s - loss: 38.9427 - val_loss: 50.4985 - 36ms/epoch - 2ms/step
Epoch 73/100
16/16 - 0s - loss: 38.8849 - val_loss: 50.4083 - 36ms/epoch - 2ms/step
Epoch 74/100
16/16 - 0s - loss: 38.8505 - val_loss: 50.2638 - 40ms/epoch - 2ms/step
Epoch 75/100
16/16 - 0s - loss: 38.7740 - val_loss: 50.2035 - 51ms/epoch - 3ms/step
Epoch 76/100
16/16 - 0s - loss: 38.7120 - val_loss: 50.3387 - 41ms/epoch - 3ms/step
Epoch 77/100
16/16 - 0s - loss: 38.7099 - val_loss: 50.2273 - 39ms/epoch - 2ms/step
Epoch 78/100
16/16 - 0s - loss: 38.6236 - val_loss: 50.1237 - 37ms/epoch - 2ms/step
Epoch 79/100
16/16 - 0s - loss: 38.5685 - val_loss: 50.0595 - 34ms/epoch - 2ms/step
Epoch 80/100
16/16 - 0s - loss: 38.5165 - val_loss: 50.0360 - 32ms/epoch - 2ms/step
Epoch 81/100
16/16 - 0s - loss: 38.4870 - val_loss: 49.9596 - 34ms/epoch - 2ms/step
Epoch 82/100
16/16 - 0s - loss: 38.4126 - val_loss: 49.9612 - 35ms/epoch - 2

16/16 - 0s - loss: 35.0258 - val_loss: 46.5613 - 35ms/epoch - 2ms/step
Epoch 68/100
16/16 - 0s - loss: 35.0549 - val_loss: 46.5839 - 35ms/epoch - 2ms/step
Epoch 69/100
16/16 - 0s - loss: 35.0277 - val_loss: 46.4693 - 34ms/epoch - 2ms/step
Epoch 70/100
16/16 - 0s - loss: 34.9917 - val_loss: 46.5823 - 34ms/epoch - 2ms/step
Epoch 71/100
16/16 - 0s - loss: 35.0079 - val_loss: 46.3557 - 33ms/epoch - 2ms/step
Epoch 72/100
16/16 - 0s - loss: 34.9193 - val_loss: 46.4611 - 36ms/epoch - 2ms/step
Epoch 73/100
16/16 - 0s - loss: 34.9268 - val_loss: 46.4205 - 35ms/epoch - 2ms/step
Epoch 74/100
16/16 - 0s - loss: 34.8996 - val_loss: 46.5678 - 35ms/epoch - 2ms/step
Epoch 75/100
16/16 - 0s - loss: 34.8955 - val_loss: 46.5117 - 41ms/epoch - 3ms/step
Epoch 76/100
16/16 - 0s - loss: 34.8502 - val_loss: 46.2925 - 37ms/epoch - 2ms/step
Epoch 77/100
16/16 - 0s - loss: 34.8136 - val_loss: 46.4595 - 35ms/epoch - 2ms/step
Epoch 78/100
16/16 - 0s - loss: 34.7938 - val_loss: 46.4220 - 37ms/epoch - 2ms/step
Epoch

Epoch 64/100
16/16 - 0s - loss: 33.2083 - val_loss: 44.6339 - 37ms/epoch - 2ms/step
Epoch 65/100
16/16 - 0s - loss: 33.2252 - val_loss: 44.5803 - 36ms/epoch - 2ms/step
Epoch 66/100
16/16 - 0s - loss: 33.2233 - val_loss: 44.6792 - 36ms/epoch - 2ms/step
Epoch 67/100
16/16 - 0s - loss: 33.1968 - val_loss: 44.5889 - 37ms/epoch - 2ms/step
Epoch 68/100
16/16 - 0s - loss: 33.1466 - val_loss: 44.6532 - 38ms/epoch - 2ms/step
Epoch 69/100
16/16 - 0s - loss: 33.2061 - val_loss: 44.5436 - 37ms/epoch - 2ms/step
Epoch 70/100
16/16 - 0s - loss: 33.1415 - val_loss: 44.6578 - 38ms/epoch - 2ms/step
Epoch 71/100
16/16 - 0s - loss: 33.1547 - val_loss: 44.6214 - 37ms/epoch - 2ms/step
Epoch 72/100
16/16 - 0s - loss: 33.1284 - val_loss: 44.5127 - 40ms/epoch - 2ms/step
Epoch 73/100
16/16 - 0s - loss: 33.1164 - val_loss: 44.5649 - 38ms/epoch - 2ms/step
Epoch 74/100
16/16 - 0s - loss: 33.1057 - val_loss: 44.5131 - 35ms/epoch - 2ms/step
Epoch 75/100
16/16 - 0s - loss: 33.0934 - val_loss: 44.4718 - 37ms/epoch - 2

16/16 - 0s - loss: 31.9109 - val_loss: 42.9940 - 37ms/epoch - 2ms/step
Epoch 61/100
16/16 - 0s - loss: 31.8807 - val_loss: 43.1942 - 38ms/epoch - 2ms/step
Epoch 62/100
16/16 - 0s - loss: 31.8694 - val_loss: 43.1452 - 40ms/epoch - 2ms/step
Epoch 63/100
16/16 - 0s - loss: 31.8914 - val_loss: 43.0234 - 37ms/epoch - 2ms/step
Epoch 64/100
16/16 - 0s - loss: 31.8843 - val_loss: 42.9895 - 37ms/epoch - 2ms/step
Epoch 65/100
16/16 - 0s - loss: 31.8349 - val_loss: 43.0258 - 37ms/epoch - 2ms/step
Epoch 66/100
16/16 - 0s - loss: 31.8204 - val_loss: 42.9520 - 37ms/epoch - 2ms/step
Epoch 67/100
16/16 - 0s - loss: 31.8444 - val_loss: 43.0141 - 35ms/epoch - 2ms/step
Epoch 68/100
16/16 - 0s - loss: 31.8037 - val_loss: 42.8113 - 35ms/epoch - 2ms/step
Epoch 69/100
16/16 - 0s - loss: 31.8389 - val_loss: 42.9239 - 36ms/epoch - 2ms/step
Epoch 70/100
16/16 - 0s - loss: 31.8143 - val_loss: 42.9277 - 38ms/epoch - 2ms/step
Epoch 71/100
16/16 - 0s - loss: 31.7624 - val_loss: 42.8594 - 41ms/epoch - 3ms/step
Epoch

Epoch 57/100
16/16 - 0s - loss: 30.8587 - val_loss: 42.1582 - 38ms/epoch - 2ms/step
Epoch 58/100
16/16 - 0s - loss: 30.7718 - val_loss: 42.1146 - 37ms/epoch - 2ms/step
Epoch 59/100
16/16 - 0s - loss: 30.7806 - val_loss: 42.1387 - 38ms/epoch - 2ms/step
Epoch 60/100
16/16 - 0s - loss: 30.8328 - val_loss: 42.2393 - 41ms/epoch - 3ms/step
Epoch 61/100
16/16 - 0s - loss: 30.7743 - val_loss: 42.1731 - 38ms/epoch - 2ms/step
Epoch 62/100
16/16 - 0s - loss: 30.7906 - val_loss: 42.2308 - 38ms/epoch - 2ms/step
Epoch 63/100
16/16 - 0s - loss: 30.7419 - val_loss: 42.2066 - 36ms/epoch - 2ms/step
Epoch 64/100
16/16 - 0s - loss: 30.7148 - val_loss: 42.1409 - 36ms/epoch - 2ms/step
Epoch 65/100
16/16 - 0s - loss: 30.7334 - val_loss: 42.0802 - 39ms/epoch - 2ms/step
Epoch 66/100
16/16 - 0s - loss: 30.7019 - val_loss: 42.0983 - 37ms/epoch - 2ms/step
Epoch 67/100
16/16 - 0s - loss: 30.7509 - val_loss: 42.1563 - 39ms/epoch - 2ms/step
Epoch 68/100
16/16 - 0s - loss: 30.6853 - val_loss: 42.1253 - 36ms/epoch - 2

16/16 - 0s - loss: 30.1377 - val_loss: 41.8366 - 36ms/epoch - 2ms/step
Epoch 54/100
16/16 - 0s - loss: 30.1052 - val_loss: 41.9092 - 37ms/epoch - 2ms/step
Epoch 55/100
16/16 - 0s - loss: 30.1350 - val_loss: 41.7809 - 36ms/epoch - 2ms/step
Epoch 56/100
16/16 - 0s - loss: 30.0948 - val_loss: 41.7407 - 36ms/epoch - 2ms/step
Epoch 57/100
16/16 - 0s - loss: 30.0913 - val_loss: 41.7666 - 38ms/epoch - 2ms/step
Epoch 58/100
16/16 - 0s - loss: 30.1167 - val_loss: 41.8446 - 36ms/epoch - 2ms/step
Epoch 59/100
16/16 - 0s - loss: 30.2033 - val_loss: 41.6681 - 38ms/epoch - 2ms/step
Epoch 60/100
16/16 - 0s - loss: 30.1821 - val_loss: 41.7650 - 36ms/epoch - 2ms/step
Epoch 61/100
16/16 - 0s - loss: 30.0744 - val_loss: 41.7610 - 38ms/epoch - 2ms/step
Epoch 62/100
16/16 - 0s - loss: 30.0817 - val_loss: 41.7674 - 38ms/epoch - 2ms/step
Epoch 63/100
16/16 - 0s - loss: 30.0885 - val_loss: 41.6721 - 38ms/epoch - 2ms/step
Epoch 64/100
16/16 - 0s - loss: 30.1328 - val_loss: 41.8383 - 35ms/epoch - 2ms/step
Epoch

Epoch 50/100
16/16 - 0s - loss: 29.6405 - val_loss: 41.4225 - 40ms/epoch - 2ms/step
Epoch 51/100
16/16 - 0s - loss: 29.6838 - val_loss: 41.1881 - 39ms/epoch - 2ms/step
Epoch 52/100
16/16 - 0s - loss: 29.6136 - val_loss: 41.2647 - 35ms/epoch - 2ms/step
Epoch 53/100
16/16 - 0s - loss: 29.6463 - val_loss: 41.3542 - 35ms/epoch - 2ms/step
Epoch 54/100
16/16 - 0s - loss: 29.6408 - val_loss: 41.2888 - 37ms/epoch - 2ms/step
Epoch 55/100
16/16 - 0s - loss: 29.6165 - val_loss: 41.1924 - 35ms/epoch - 2ms/step
Epoch 56/100
16/16 - 0s - loss: 29.6257 - val_loss: 41.3129 - 36ms/epoch - 2ms/step
Epoch 57/100
16/16 - 0s - loss: 29.5816 - val_loss: 41.2332 - 39ms/epoch - 2ms/step
Epoch 58/100
16/16 - 0s - loss: 29.6109 - val_loss: 41.2856 - 36ms/epoch - 2ms/step
Epoch 59/100
16/16 - 0s - loss: 29.5828 - val_loss: 41.2290 - 37ms/epoch - 2ms/step
Epoch 60/100
16/16 - 0s - loss: 29.6575 - val_loss: 41.1369 - 37ms/epoch - 2ms/step
Epoch 61/100
16/16 - 0s - loss: 29.5922 - val_loss: 41.2774 - 35ms/epoch - 2

16/16 - 0s - loss: 29.2312 - val_loss: 40.9309 - 34ms/epoch - 2ms/step
Epoch 47/100
16/16 - 0s - loss: 29.2462 - val_loss: 40.8866 - 35ms/epoch - 2ms/step
Epoch 48/100
16/16 - 0s - loss: 29.2225 - val_loss: 40.8898 - 38ms/epoch - 2ms/step
Epoch 49/100
16/16 - 0s - loss: 29.2179 - val_loss: 40.8273 - 36ms/epoch - 2ms/step
Epoch 50/100
16/16 - 0s - loss: 29.2715 - val_loss: 40.8241 - 38ms/epoch - 2ms/step
Epoch 51/100
16/16 - 0s - loss: 29.1895 - val_loss: 40.8833 - 38ms/epoch - 2ms/step
Epoch 52/100
16/16 - 0s - loss: 29.2139 - val_loss: 40.8999 - 35ms/epoch - 2ms/step
Epoch 53/100
16/16 - 0s - loss: 29.2537 - val_loss: 40.8620 - 36ms/epoch - 2ms/step
Epoch 54/100
16/16 - 0s - loss: 29.2030 - val_loss: 40.8357 - 36ms/epoch - 2ms/step
Epoch 55/100
16/16 - 0s - loss: 29.2296 - val_loss: 40.9457 - 37ms/epoch - 2ms/step
Epoch 56/100
16/16 - 0s - loss: 29.2096 - val_loss: 40.8500 - 36ms/epoch - 2ms/step
Epoch 57/100
16/16 - 0s - loss: 29.1867 - val_loss: 40.8017 - 35ms/epoch - 2ms/step
Epoch

Epoch 43/100
16/16 - 0s - loss: 28.9483 - val_loss: 40.5417 - 37ms/epoch - 2ms/step
Epoch 44/100
16/16 - 0s - loss: 28.9426 - val_loss: 40.6691 - 37ms/epoch - 2ms/step
Epoch 45/100
16/16 - 0s - loss: 28.9073 - val_loss: 40.6777 - 35ms/epoch - 2ms/step
Epoch 46/100
16/16 - 0s - loss: 28.8943 - val_loss: 40.5950 - 34ms/epoch - 2ms/step
Epoch 47/100
16/16 - 0s - loss: 28.9355 - val_loss: 40.7078 - 37ms/epoch - 2ms/step
Epoch 48/100
16/16 - 0s - loss: 28.9337 - val_loss: 40.5507 - 35ms/epoch - 2ms/step
Epoch 49/100
16/16 - 0s - loss: 28.9715 - val_loss: 40.7151 - 34ms/epoch - 2ms/step
Epoch 50/100
16/16 - 0s - loss: 28.9209 - val_loss: 40.5886 - 38ms/epoch - 2ms/step
Epoch 51/100
16/16 - 0s - loss: 28.8657 - val_loss: 40.6013 - 36ms/epoch - 2ms/step
Epoch 52/100
16/16 - 0s - loss: 28.8928 - val_loss: 40.4550 - 36ms/epoch - 2ms/step
Epoch 53/100
16/16 - 0s - loss: 28.9053 - val_loss: 40.6063 - 36ms/epoch - 2ms/step
Epoch 54/100
16/16 - 0s - loss: 28.8668 - val_loss: 40.6723 - 37ms/epoch - 2

16/16 - 0s - loss: 28.7395 - val_loss: 40.4242 - 36ms/epoch - 2ms/step
Epoch 40/100
16/16 - 0s - loss: 28.6820 - val_loss: 40.4130 - 39ms/epoch - 2ms/step
Epoch 41/100
16/16 - 0s - loss: 28.6805 - val_loss: 40.4143 - 37ms/epoch - 2ms/step
Epoch 42/100
16/16 - 0s - loss: 28.6862 - val_loss: 40.5580 - 36ms/epoch - 2ms/step
Epoch 43/100
16/16 - 0s - loss: 28.7570 - val_loss: 40.3373 - 37ms/epoch - 2ms/step
Epoch 44/100
16/16 - 0s - loss: 28.6818 - val_loss: 40.4488 - 38ms/epoch - 2ms/step
Epoch 45/100
16/16 - 0s - loss: 28.6690 - val_loss: 40.5156 - 37ms/epoch - 2ms/step
Epoch 46/100
16/16 - 0s - loss: 28.6925 - val_loss: 40.4062 - 33ms/epoch - 2ms/step
Epoch 47/100
16/16 - 0s - loss: 28.6772 - val_loss: 40.4616 - 35ms/epoch - 2ms/step
Epoch 48/100
16/16 - 0s - loss: 28.6986 - val_loss: 40.4988 - 35ms/epoch - 2ms/step
Epoch 49/100
16/16 - 0s - loss: 28.6966 - val_loss: 40.4950 - 36ms/epoch - 2ms/step
Epoch 50/100
16/16 - 0s - loss: 28.6851 - val_loss: 40.3765 - 35ms/epoch - 2ms/step
Epoch

Epoch 36/100
16/16 - 0s - loss: 28.5333 - val_loss: 40.4007 - 35ms/epoch - 2ms/step
Epoch 37/100
16/16 - 0s - loss: 28.5277 - val_loss: 40.3945 - 35ms/epoch - 2ms/step
Epoch 38/100
16/16 - 0s - loss: 28.5581 - val_loss: 40.2903 - 36ms/epoch - 2ms/step
Epoch 39/100
16/16 - 0s - loss: 28.5010 - val_loss: 40.3546 - 35ms/epoch - 2ms/step
Epoch 40/100
16/16 - 0s - loss: 28.5255 - val_loss: 40.3459 - 35ms/epoch - 2ms/step
Epoch 41/100
16/16 - 0s - loss: 28.5452 - val_loss: 40.3116 - 38ms/epoch - 2ms/step
Epoch 42/100
16/16 - 0s - loss: 28.5150 - val_loss: 40.4179 - 34ms/epoch - 2ms/step
Epoch 43/100
16/16 - 0s - loss: 28.4885 - val_loss: 40.3122 - 34ms/epoch - 2ms/step
Epoch 44/100
16/16 - 0s - loss: 28.5815 - val_loss: 40.4407 - 35ms/epoch - 2ms/step
Epoch 45/100
16/16 - 0s - loss: 28.5633 - val_loss: 40.2509 - 36ms/epoch - 2ms/step
Epoch 46/100
16/16 - 0s - loss: 28.5074 - val_loss: 40.2436 - 37ms/epoch - 2ms/step
Epoch 47/100
16/16 - 0s - loss: 28.4926 - val_loss: 40.3649 - 36ms/epoch - 2

16/16 - 0s - loss: 28.4110 - val_loss: 40.3329 - 34ms/epoch - 2ms/step
Epoch 33/100
16/16 - 0s - loss: 28.4281 - val_loss: 40.3445 - 37ms/epoch - 2ms/step
Epoch 34/100
16/16 - 0s - loss: 28.3934 - val_loss: 40.1942 - 39ms/epoch - 2ms/step
Epoch 35/100
16/16 - 0s - loss: 28.3643 - val_loss: 40.3210 - 35ms/epoch - 2ms/step
Epoch 36/100
16/16 - 0s - loss: 28.4208 - val_loss: 40.2909 - 32ms/epoch - 2ms/step
Epoch 37/100
16/16 - 0s - loss: 28.3809 - val_loss: 40.3279 - 32ms/epoch - 2ms/step
Epoch 38/100
16/16 - 0s - loss: 28.3757 - val_loss: 40.2840 - 32ms/epoch - 2ms/step
Epoch 39/100
16/16 - 0s - loss: 28.3892 - val_loss: 40.2352 - 35ms/epoch - 2ms/step
Epoch 40/100
16/16 - 0s - loss: 28.4106 - val_loss: 40.2884 - 37ms/epoch - 2ms/step
Epoch 41/100
16/16 - 0s - loss: 28.3815 - val_loss: 40.2338 - 35ms/epoch - 2ms/step
Epoch 42/100
16/16 - 0s - loss: 28.3706 - val_loss: 40.3542 - 37ms/epoch - 2ms/step
Epoch 43/100
16/16 - 0s - loss: 28.4069 - val_loss: 40.3825 - 35ms/epoch - 2ms/step
Epoch

Epoch 29/100
16/16 - 0s - loss: 28.3150 - val_loss: 40.1649 - 37ms/epoch - 2ms/step
Epoch 30/100
16/16 - 0s - loss: 28.2748 - val_loss: 40.3539 - 35ms/epoch - 2ms/step
Epoch 31/100
16/16 - 0s - loss: 28.2573 - val_loss: 40.2680 - 37ms/epoch - 2ms/step
Epoch 32/100
16/16 - 0s - loss: 28.3015 - val_loss: 40.2405 - 38ms/epoch - 2ms/step
Epoch 33/100
16/16 - 0s - loss: 28.2447 - val_loss: 40.3448 - 37ms/epoch - 2ms/step
Epoch 34/100
16/16 - 0s - loss: 28.3123 - val_loss: 40.4185 - 40ms/epoch - 2ms/step
Epoch 35/100
16/16 - 0s - loss: 28.2633 - val_loss: 40.3617 - 45ms/epoch - 3ms/step
Epoch 36/100
16/16 - 0s - loss: 28.2803 - val_loss: 40.1673 - 44ms/epoch - 3ms/step
Epoch 37/100
16/16 - 0s - loss: 28.2773 - val_loss: 40.2126 - 39ms/epoch - 2ms/step
Epoch 38/100
16/16 - 0s - loss: 28.2654 - val_loss: 40.2494 - 38ms/epoch - 2ms/step
Epoch 39/100
16/16 - 0s - loss: 28.2375 - val_loss: 40.2811 - 41ms/epoch - 3ms/step
Epoch 40/100
16/16 - 0s - loss: 28.3087 - val_loss: 40.2726 - 37ms/epoch - 2

Epoch 26/100
16/16 - 0s - loss: 28.1947 - val_loss: 40.3969 - 35ms/epoch - 2ms/step
Epoch 27/100
16/16 - 0s - loss: 28.1560 - val_loss: 40.1838 - 34ms/epoch - 2ms/step
Epoch 28/100
16/16 - 0s - loss: 28.1471 - val_loss: 40.2079 - 37ms/epoch - 2ms/step
Epoch 29/100
16/16 - 0s - loss: 28.1871 - val_loss: 40.1817 - 36ms/epoch - 2ms/step
Epoch 30/100
16/16 - 0s - loss: 28.1535 - val_loss: 40.3179 - 35ms/epoch - 2ms/step
Epoch 31/100
16/16 - 0s - loss: 28.1441 - val_loss: 40.2460 - 38ms/epoch - 2ms/step
Epoch 32/100
16/16 - 0s - loss: 28.1718 - val_loss: 40.3473 - 36ms/epoch - 2ms/step
Epoch 33/100
16/16 - 0s - loss: 28.2135 - val_loss: 40.2377 - 39ms/epoch - 2ms/step
Epoch 34/100
16/16 - 0s - loss: 28.1645 - val_loss: 40.2173 - 39ms/epoch - 2ms/step
Epoch 35/100
16/16 - 0s - loss: 28.2034 - val_loss: 40.1289 - 37ms/epoch - 2ms/step
Epoch 36/100
16/16 - 0s - loss: 28.1378 - val_loss: 40.2716 - 36ms/epoch - 2ms/step
Epoch 37/100
16/16 - 0s - loss: 28.1650 - val_loss: 40.3278 - 35ms/epoch - 2

16/16 - 0s - loss: 28.0607 - val_loss: 40.2555 - 37ms/epoch - 2ms/step
Epoch 23/100
16/16 - 0s - loss: 28.0884 - val_loss: 40.2326 - 37ms/epoch - 2ms/step
Epoch 24/100
16/16 - 0s - loss: 28.0395 - val_loss: 40.0691 - 35ms/epoch - 2ms/step
Epoch 25/100
16/16 - 0s - loss: 28.0481 - val_loss: 40.1180 - 41ms/epoch - 3ms/step
Epoch 26/100
16/16 - 0s - loss: 28.0913 - val_loss: 40.2139 - 36ms/epoch - 2ms/step
Epoch 27/100
16/16 - 0s - loss: 28.0309 - val_loss: 40.2053 - 37ms/epoch - 2ms/step
Epoch 28/100
16/16 - 0s - loss: 28.0733 - val_loss: 40.1223 - 35ms/epoch - 2ms/step
Epoch 29/100
16/16 - 0s - loss: 28.0524 - val_loss: 40.2897 - 39ms/epoch - 2ms/step
Epoch 30/100
16/16 - 0s - loss: 28.1092 - val_loss: 40.1146 - 38ms/epoch - 2ms/step
Epoch 31/100
16/16 - 0s - loss: 28.0483 - val_loss: 40.2838 - 44ms/epoch - 3ms/step
Epoch 32/100
16/16 - 0s - loss: 28.0454 - val_loss: 40.1729 - 44ms/epoch - 3ms/step
Epoch 33/100
16/16 - 0s - loss: 28.0325 - val_loss: 40.1844 - 39ms/epoch - 2ms/step
Epoch

Epoch 19/100
16/16 - 0s - loss: 27.9659 - val_loss: 39.9964 - 36ms/epoch - 2ms/step
Epoch 20/100
16/16 - 0s - loss: 27.9443 - val_loss: 40.1546 - 37ms/epoch - 2ms/step
Epoch 21/100
16/16 - 0s - loss: 27.9692 - val_loss: 40.0362 - 35ms/epoch - 2ms/step
Epoch 22/100
16/16 - 0s - loss: 27.9666 - val_loss: 40.1597 - 37ms/epoch - 2ms/step
Epoch 23/100
16/16 - 0s - loss: 27.9022 - val_loss: 40.0606 - 37ms/epoch - 2ms/step
Epoch 24/100
16/16 - 0s - loss: 27.9121 - val_loss: 40.0321 - 35ms/epoch - 2ms/step
Epoch 25/100
16/16 - 0s - loss: 27.9533 - val_loss: 39.9532 - 34ms/epoch - 2ms/step
Epoch 26/100
16/16 - 0s - loss: 27.9914 - val_loss: 40.1508 - 36ms/epoch - 2ms/step
Epoch 27/100
16/16 - 0s - loss: 27.9467 - val_loss: 40.0229 - 36ms/epoch - 2ms/step
Epoch 28/100
16/16 - 0s - loss: 27.9150 - val_loss: 40.1214 - 38ms/epoch - 2ms/step
Epoch 29/100
16/16 - 0s - loss: 27.9307 - val_loss: 40.0679 - 35ms/epoch - 2ms/step
Epoch 30/100
16/16 - 0s - loss: 27.9420 - val_loss: 40.0997 - 38ms/epoch - 2

16/16 - 0s - loss: 27.8295 - val_loss: 40.0162 - 37ms/epoch - 2ms/step
Epoch 16/100
16/16 - 0s - loss: 27.8256 - val_loss: 40.1216 - 35ms/epoch - 2ms/step
Epoch 17/100
16/16 - 0s - loss: 27.8174 - val_loss: 40.1040 - 35ms/epoch - 2ms/step
Epoch 18/100
16/16 - 0s - loss: 27.8267 - val_loss: 40.0916 - 31ms/epoch - 2ms/step
Epoch 19/100
16/16 - 0s - loss: 27.8272 - val_loss: 40.0014 - 34ms/epoch - 2ms/step
Epoch 20/100
16/16 - 0s - loss: 27.8148 - val_loss: 40.0663 - 34ms/epoch - 2ms/step
Epoch 21/100
16/16 - 0s - loss: 27.8914 - val_loss: 39.9805 - 35ms/epoch - 2ms/step
Epoch 22/100
16/16 - 0s - loss: 27.8208 - val_loss: 40.0867 - 34ms/epoch - 2ms/step
Epoch 23/100
16/16 - 0s - loss: 27.9752 - val_loss: 40.2668 - 34ms/epoch - 2ms/step
Epoch 24/100
16/16 - 0s - loss: 27.8168 - val_loss: 40.0774 - 35ms/epoch - 2ms/step
Epoch 25/100
16/16 - 0s - loss: 27.8039 - val_loss: 39.9578 - 34ms/epoch - 2ms/step
Epoch 26/100
16/16 - 0s - loss: 27.8045 - val_loss: 40.0044 - 35ms/epoch - 2ms/step
Epoch

Epoch 12/100
16/16 - 0s - loss: 27.7163 - val_loss: 40.1439 - 35ms/epoch - 2ms/step
Epoch 13/100
16/16 - 0s - loss: 27.7192 - val_loss: 40.1728 - 34ms/epoch - 2ms/step
Epoch 14/100
16/16 - 0s - loss: 27.7498 - val_loss: 40.1281 - 36ms/epoch - 2ms/step
Epoch 15/100
16/16 - 0s - loss: 27.7523 - val_loss: 39.9740 - 37ms/epoch - 2ms/step
Epoch 16/100
16/16 - 0s - loss: 27.7378 - val_loss: 40.0023 - 35ms/epoch - 2ms/step
Epoch 17/100
16/16 - 0s - loss: 27.7251 - val_loss: 40.1427 - 36ms/epoch - 2ms/step
Epoch 18/100
16/16 - 0s - loss: 27.7256 - val_loss: 40.0731 - 34ms/epoch - 2ms/step
Epoch 19/100
16/16 - 0s - loss: 27.7079 - val_loss: 40.1083 - 38ms/epoch - 2ms/step
Epoch 20/100
16/16 - 0s - loss: 27.7660 - val_loss: 40.1156 - 35ms/epoch - 2ms/step
Epoch 21/100
16/16 - 0s - loss: 27.7089 - val_loss: 40.0669 - 36ms/epoch - 2ms/step
Epoch 22/100
16/16 - 0s - loss: 27.7614 - val_loss: 40.1593 - 35ms/epoch - 2ms/step
Epoch 23/100
16/16 - 0s - loss: 27.7764 - val_loss: 40.1772 - 35ms/epoch - 2

16/16 - 0s - loss: 27.6153 - val_loss: 40.2320 - 35ms/epoch - 2ms/step
Epoch 9/100
16/16 - 0s - loss: 27.6739 - val_loss: 40.1736 - 36ms/epoch - 2ms/step
Epoch 10/100
16/16 - 0s - loss: 27.6183 - val_loss: 40.1890 - 71ms/epoch - 4ms/step
Epoch 11/100
16/16 - 0s - loss: 27.6245 - val_loss: 40.3409 - 37ms/epoch - 2ms/step
Epoch 12/100
16/16 - 0s - loss: 27.6135 - val_loss: 40.3651 - 34ms/epoch - 2ms/step
Epoch 13/100
16/16 - 0s - loss: 27.6096 - val_loss: 40.2621 - 33ms/epoch - 2ms/step
Epoch 14/100
16/16 - 0s - loss: 27.6512 - val_loss: 40.1971 - 34ms/epoch - 2ms/step
Epoch 15/100
16/16 - 0s - loss: 27.6319 - val_loss: 40.2282 - 35ms/epoch - 2ms/step
Epoch 16/100
16/16 - 0s - loss: 27.5916 - val_loss: 40.1919 - 35ms/epoch - 2ms/step
Epoch 17/100
16/16 - 0s - loss: 27.6309 - val_loss: 40.1826 - 34ms/epoch - 2ms/step
Epoch 18/100
16/16 - 0s - loss: 27.6738 - val_loss: 40.2592 - 34ms/epoch - 2ms/step
Epoch 19/100
16/16 - 0s - loss: 27.6172 - val_loss: 40.2620 - 36ms/epoch - 2ms/step
Epoch 

Epoch 5/100
16/16 - 0s - loss: 27.5815 - val_loss: 40.1979 - 35ms/epoch - 2ms/step
Epoch 6/100
16/16 - 0s - loss: 27.5702 - val_loss: 40.2905 - 35ms/epoch - 2ms/step
Epoch 7/100
16/16 - 0s - loss: 27.5580 - val_loss: 40.2627 - 36ms/epoch - 2ms/step
Epoch 8/100
16/16 - 0s - loss: 27.5849 - val_loss: 40.1516 - 35ms/epoch - 2ms/step
Epoch 9/100
16/16 - 0s - loss: 27.5475 - val_loss: 40.2446 - 36ms/epoch - 2ms/step
Epoch 10/100
16/16 - 0s - loss: 27.5447 - val_loss: 40.2687 - 35ms/epoch - 2ms/step
Epoch 11/100
16/16 - 0s - loss: 27.5524 - val_loss: 40.3859 - 36ms/epoch - 2ms/step
Epoch 12/100
16/16 - 0s - loss: 27.5117 - val_loss: 40.2181 - 35ms/epoch - 2ms/step
Epoch 13/100
16/16 - 0s - loss: 27.6070 - val_loss: 40.1235 - 36ms/epoch - 2ms/step
Epoch 14/100
16/16 - 0s - loss: 27.5412 - val_loss: 40.2300 - 34ms/epoch - 2ms/step
Epoch 15/100
16/16 - 0s - loss: 27.5552 - val_loss: 40.2339 - 35ms/epoch - 2ms/step
Epoch 16/100
16/16 - 0s - loss: 27.5934 - val_loss: 40.3854 - 37ms/epoch - 2ms/st

16/16 - 0s - loss: 27.4949 - val_loss: 40.1870 - 73ms/epoch - 5ms/step
Epoch 2/100
16/16 - 0s - loss: 27.5041 - val_loss: 40.3176 - 33ms/epoch - 2ms/step
Epoch 3/100
16/16 - 0s - loss: 27.5052 - val_loss: 40.2817 - 36ms/epoch - 2ms/step
Epoch 4/100
16/16 - 0s - loss: 27.5652 - val_loss: 40.3402 - 35ms/epoch - 2ms/step
Epoch 5/100
16/16 - 0s - loss: 27.5325 - val_loss: 40.2760 - 40ms/epoch - 2ms/step
Epoch 6/100
16/16 - 0s - loss: 27.5688 - val_loss: 40.2294 - 37ms/epoch - 2ms/step
Epoch 7/100
16/16 - 0s - loss: 27.4921 - val_loss: 40.0739 - 34ms/epoch - 2ms/step
Epoch 8/100
16/16 - 0s - loss: 27.5538 - val_loss: 40.2061 - 36ms/epoch - 2ms/step
Epoch 9/100
16/16 - 0s - loss: 27.5245 - val_loss: 40.2416 - 37ms/epoch - 2ms/step
Epoch 10/100
16/16 - 0s - loss: 27.5175 - val_loss: 40.3196 - 35ms/epoch - 2ms/step
Epoch 11/100
16/16 - 0s - loss: 27.5292 - val_loss: 40.2192 - 34ms/epoch - 2ms/step
Epoch 12/100
16/16 - 0s - loss: 27.5137 - val_loss: 40.2070 - 36ms/epoch - 2ms/step
Epoch 13/100


Epoch 99/100
16/16 - 0s - loss: 27.5565 - val_loss: 40.3562 - 40ms/epoch - 2ms/step
Epoch 100/100
16/16 - 0s - loss: 27.4454 - val_loss: 40.2902 - 36ms/epoch - 2ms/step
Accuracy: 37.65026092529297 
 Error: -3665.026092529297
10/10 [==============================] - 0s 997us/step
Iteration 27:

Epoch 1/100
16/16 - 0s - loss: 27.4446 - val_loss: 40.1749 - 77ms/epoch - 5ms/step
Epoch 2/100
16/16 - 0s - loss: 27.4500 - val_loss: 40.2049 - 37ms/epoch - 2ms/step
Epoch 3/100
16/16 - 0s - loss: 27.4665 - val_loss: 40.1179 - 35ms/epoch - 2ms/step
Epoch 4/100
16/16 - 0s - loss: 27.4671 - val_loss: 40.0951 - 35ms/epoch - 2ms/step
Epoch 5/100
16/16 - 0s - loss: 27.4485 - val_loss: 40.2343 - 37ms/epoch - 2ms/step
Epoch 6/100
16/16 - 0s - loss: 27.4380 - val_loss: 40.2302 - 39ms/epoch - 2ms/step
Epoch 7/100
16/16 - 0s - loss: 27.4588 - val_loss: 40.2496 - 39ms/epoch - 2ms/step
Epoch 8/100
16/16 - 0s - loss: 27.4487 - val_loss: 40.1404 - 37ms/epoch - 2ms/step
Epoch 9/100
16/16 - 0s - loss: 27.4390 - 

16/16 - 0s - loss: 27.4739 - val_loss: 40.3372 - 35ms/epoch - 2ms/step
Epoch 96/100
16/16 - 0s - loss: 27.4243 - val_loss: 40.2166 - 33ms/epoch - 2ms/step
Epoch 97/100
16/16 - 0s - loss: 27.4129 - val_loss: 40.1655 - 35ms/epoch - 2ms/step
Epoch 98/100
16/16 - 0s - loss: 27.4669 - val_loss: 40.1277 - 35ms/epoch - 2ms/step
Epoch 99/100
16/16 - 0s - loss: 27.4220 - val_loss: 40.2687 - 32ms/epoch - 2ms/step
Epoch 100/100
16/16 - 0s - loss: 27.4188 - val_loss: 40.2368 - 33ms/epoch - 2ms/step
Accuracy: 37.63310241699219 
 Error: -3663.3102416992188
10/10 [==============================] - 0s 997us/step
Iteration 28:

Epoch 1/100
16/16 - 0s - loss: 27.4179 - val_loss: 40.1404 - 73ms/epoch - 5ms/step
Epoch 2/100
16/16 - 0s - loss: 27.4436 - val_loss: 40.1067 - 36ms/epoch - 2ms/step
Epoch 3/100
16/16 - 0s - loss: 27.4663 - val_loss: 40.0916 - 35ms/epoch - 2ms/step
Epoch 4/100
16/16 - 0s - loss: 27.4674 - val_loss: 40.1525 - 35ms/epoch - 2ms/step
Epoch 5/100
16/16 - 0s - loss: 27.4051 - val_loss

Epoch 92/100
16/16 - 0s - loss: 27.3762 - val_loss: 40.1935 - 33ms/epoch - 2ms/step
Epoch 93/100
16/16 - 0s - loss: 27.4101 - val_loss: 40.0876 - 37ms/epoch - 2ms/step
Epoch 94/100
16/16 - 0s - loss: 27.4444 - val_loss: 40.0565 - 35ms/epoch - 2ms/step
Epoch 95/100
16/16 - 0s - loss: 27.3989 - val_loss: 40.1681 - 33ms/epoch - 2ms/step
Epoch 96/100
16/16 - 0s - loss: 27.3918 - val_loss: 40.1038 - 34ms/epoch - 2ms/step
Epoch 97/100
16/16 - 0s - loss: 27.3987 - val_loss: 40.0730 - 35ms/epoch - 2ms/step
Epoch 98/100
16/16 - 0s - loss: 27.4329 - val_loss: 40.1415 - 33ms/epoch - 2ms/step
Epoch 99/100
16/16 - 0s - loss: 27.4256 - val_loss: 40.2979 - 33ms/epoch - 2ms/step
Epoch 100/100
16/16 - 0s - loss: 27.3867 - val_loss: 40.1576 - 34ms/epoch - 2ms/step
Accuracy: 37.52718734741211 
 Error: -3652.718734741211
10/10 [==============================] - 0s 889us/step
Iteration 29:

Epoch 1/100
16/16 - 0s - loss: 27.3893 - val_loss: 40.1493 - 71ms/epoch - 4ms/step
Epoch 2/100
16/16 - 0s - loss: 27.

16/16 - 0s - loss: 27.3946 - val_loss: 40.0270 - 34ms/epoch - 2ms/step
Epoch 89/100
16/16 - 0s - loss: 27.3626 - val_loss: 40.1625 - 34ms/epoch - 2ms/step
Epoch 90/100
16/16 - 0s - loss: 27.3671 - val_loss: 40.2353 - 40ms/epoch - 2ms/step
Epoch 91/100
16/16 - 0s - loss: 27.3729 - val_loss: 40.0751 - 39ms/epoch - 2ms/step
Epoch 92/100
16/16 - 0s - loss: 27.3880 - val_loss: 39.9520 - 37ms/epoch - 2ms/step
Epoch 93/100
16/16 - 0s - loss: 27.3893 - val_loss: 40.0733 - 35ms/epoch - 2ms/step
Epoch 94/100
16/16 - 0s - loss: 27.3576 - val_loss: 40.1146 - 36ms/epoch - 2ms/step
Epoch 95/100
16/16 - 0s - loss: 27.3593 - val_loss: 40.1209 - 35ms/epoch - 2ms/step
Epoch 96/100
16/16 - 0s - loss: 27.3710 - val_loss: 40.1915 - 35ms/epoch - 2ms/step
Epoch 97/100
16/16 - 0s - loss: 27.3821 - val_loss: 40.0446 - 37ms/epoch - 2ms/step
Epoch 98/100
16/16 - 0s - loss: 27.3746 - val_loss: 40.0907 - 33ms/epoch - 2ms/step
Epoch 99/100
16/16 - 0s - loss: 27.3757 - val_loss: 40.2508 - 35ms/epoch - 2ms/step
Epoch

Epoch 85/100
16/16 - 0s - loss: 27.3401 - val_loss: 40.0796 - 35ms/epoch - 2ms/step
Epoch 86/100
16/16 - 0s - loss: 27.3964 - val_loss: 40.1078 - 33ms/epoch - 2ms/step
Epoch 87/100
16/16 - 0s - loss: 27.3536 - val_loss: 39.9459 - 36ms/epoch - 2ms/step
Epoch 88/100
16/16 - 0s - loss: 27.3538 - val_loss: 39.9379 - 32ms/epoch - 2ms/step
Epoch 89/100
16/16 - 0s - loss: 27.3144 - val_loss: 39.9766 - 33ms/epoch - 2ms/step
Epoch 90/100
16/16 - 0s - loss: 27.3449 - val_loss: 40.0293 - 34ms/epoch - 2ms/step
Epoch 91/100
16/16 - 0s - loss: 27.3408 - val_loss: 39.9917 - 35ms/epoch - 2ms/step
Epoch 92/100
16/16 - 0s - loss: 27.3948 - val_loss: 40.0895 - 36ms/epoch - 2ms/step
Epoch 93/100
16/16 - 0s - loss: 27.3652 - val_loss: 39.8835 - 35ms/epoch - 2ms/step
Epoch 94/100
16/16 - 0s - loss: 27.3274 - val_loss: 40.0476 - 33ms/epoch - 2ms/step
Epoch 95/100
16/16 - 0s - loss: 27.3395 - val_loss: 40.0532 - 35ms/epoch - 2ms/step
Epoch 96/100
16/16 - 0s - loss: 27.3470 - val_loss: 40.0488 - 33ms/epoch - 2

16/16 - 0s - loss: 27.3344 - val_loss: 40.0566 - 34ms/epoch - 2ms/step
Epoch 82/100
16/16 - 0s - loss: 27.2950 - val_loss: 39.9408 - 37ms/epoch - 2ms/step
Epoch 83/100
16/16 - 0s - loss: 27.2890 - val_loss: 39.9135 - 33ms/epoch - 2ms/step
Epoch 84/100
16/16 - 0s - loss: 27.2925 - val_loss: 40.0017 - 35ms/epoch - 2ms/step
Epoch 85/100
16/16 - 0s - loss: 27.2717 - val_loss: 40.0456 - 34ms/epoch - 2ms/step
Epoch 86/100
16/16 - 0s - loss: 27.3059 - val_loss: 40.0138 - 34ms/epoch - 2ms/step
Epoch 87/100
16/16 - 0s - loss: 27.2972 - val_loss: 39.9556 - 33ms/epoch - 2ms/step
Epoch 88/100
16/16 - 0s - loss: 27.2756 - val_loss: 39.9919 - 33ms/epoch - 2ms/step
Epoch 89/100
16/16 - 0s - loss: 27.3086 - val_loss: 40.0241 - 34ms/epoch - 2ms/step
Epoch 90/100
16/16 - 0s - loss: 27.2878 - val_loss: 40.0364 - 33ms/epoch - 2ms/step
Epoch 91/100
16/16 - 0s - loss: 27.3647 - val_loss: 40.0058 - 33ms/epoch - 2ms/step
Epoch 92/100
16/16 - 0s - loss: 27.3028 - val_loss: 39.9396 - 34ms/epoch - 2ms/step
Epoch

Epoch 78/100
16/16 - 0s - loss: 27.2629 - val_loss: 39.9664 - 33ms/epoch - 2ms/step
Epoch 79/100
16/16 - 0s - loss: 27.3331 - val_loss: 40.0569 - 33ms/epoch - 2ms/step
Epoch 80/100
16/16 - 0s - loss: 27.2734 - val_loss: 39.9336 - 34ms/epoch - 2ms/step
Epoch 81/100
16/16 - 0s - loss: 27.2650 - val_loss: 39.9150 - 36ms/epoch - 2ms/step
Epoch 82/100
16/16 - 0s - loss: 27.2620 - val_loss: 39.8827 - 35ms/epoch - 2ms/step
Epoch 83/100
16/16 - 0s - loss: 27.3000 - val_loss: 39.9488 - 34ms/epoch - 2ms/step
Epoch 84/100
16/16 - 0s - loss: 27.2479 - val_loss: 39.9314 - 35ms/epoch - 2ms/step
Epoch 85/100
16/16 - 0s - loss: 27.2862 - val_loss: 39.8653 - 34ms/epoch - 2ms/step
Epoch 86/100
16/16 - 0s - loss: 27.2751 - val_loss: 39.9803 - 39ms/epoch - 2ms/step
Epoch 87/100
16/16 - 0s - loss: 27.2562 - val_loss: 40.0493 - 36ms/epoch - 2ms/step
Epoch 88/100
16/16 - 0s - loss: 27.2731 - val_loss: 40.0201 - 35ms/epoch - 2ms/step
Epoch 89/100
16/16 - 0s - loss: 27.2602 - val_loss: 39.9966 - 34ms/epoch - 2

16/16 - 0s - loss: 27.2251 - val_loss: 39.9222 - 35ms/epoch - 2ms/step
Epoch 75/100
16/16 - 0s - loss: 27.2850 - val_loss: 40.1748 - 35ms/epoch - 2ms/step
Epoch 76/100
16/16 - 0s - loss: 27.2569 - val_loss: 39.8996 - 35ms/epoch - 2ms/step
Epoch 77/100
16/16 - 0s - loss: 27.2440 - val_loss: 39.8749 - 35ms/epoch - 2ms/step
Epoch 78/100
16/16 - 0s - loss: 27.2429 - val_loss: 39.9636 - 38ms/epoch - 2ms/step
Epoch 79/100
16/16 - 0s - loss: 27.2617 - val_loss: 39.9141 - 35ms/epoch - 2ms/step
Epoch 80/100
16/16 - 0s - loss: 27.2631 - val_loss: 39.9979 - 35ms/epoch - 2ms/step
Epoch 81/100
16/16 - 0s - loss: 27.2512 - val_loss: 39.9643 - 36ms/epoch - 2ms/step
Epoch 82/100
16/16 - 0s - loss: 27.2714 - val_loss: 40.0345 - 38ms/epoch - 2ms/step
Epoch 83/100
16/16 - 0s - loss: 27.2418 - val_loss: 39.9540 - 39ms/epoch - 2ms/step
Epoch 84/100
16/16 - 0s - loss: 27.3183 - val_loss: 40.0251 - 33ms/epoch - 2ms/step
Epoch 85/100
16/16 - 0s - loss: 27.2246 - val_loss: 39.9840 - 36ms/epoch - 2ms/step
Epoch

Epoch 71/100
16/16 - 0s - loss: 27.2141 - val_loss: 40.0283 - 36ms/epoch - 2ms/step
Epoch 72/100
16/16 - 0s - loss: 27.1995 - val_loss: 39.9875 - 35ms/epoch - 2ms/step
Epoch 73/100
16/16 - 0s - loss: 27.1925 - val_loss: 39.9657 - 35ms/epoch - 2ms/step
Epoch 74/100
16/16 - 0s - loss: 27.2237 - val_loss: 39.9367 - 35ms/epoch - 2ms/step
Epoch 75/100
16/16 - 0s - loss: 27.1999 - val_loss: 40.0012 - 35ms/epoch - 2ms/step
Epoch 76/100
16/16 - 0s - loss: 27.2464 - val_loss: 39.9950 - 34ms/epoch - 2ms/step
Epoch 77/100
16/16 - 0s - loss: 27.2077 - val_loss: 39.9336 - 35ms/epoch - 2ms/step
Epoch 78/100
16/16 - 0s - loss: 27.2195 - val_loss: 39.8738 - 34ms/epoch - 2ms/step
Epoch 79/100
16/16 - 0s - loss: 27.2013 - val_loss: 39.8602 - 33ms/epoch - 2ms/step
Epoch 80/100
16/16 - 0s - loss: 27.2201 - val_loss: 40.0219 - 34ms/epoch - 2ms/step
Epoch 81/100
16/16 - 0s - loss: 27.2047 - val_loss: 39.9472 - 33ms/epoch - 2ms/step
Epoch 82/100
16/16 - 0s - loss: 27.2086 - val_loss: 39.8577 - 34ms/epoch - 2

16/16 - 0s - loss: 27.2131 - val_loss: 39.9354 - 33ms/epoch - 2ms/step
Epoch 68/100
16/16 - 0s - loss: 27.2249 - val_loss: 39.9174 - 35ms/epoch - 2ms/step
Epoch 69/100
16/16 - 0s - loss: 27.2083 - val_loss: 39.9079 - 36ms/epoch - 2ms/step
Epoch 70/100
16/16 - 0s - loss: 27.1908 - val_loss: 39.9005 - 37ms/epoch - 2ms/step
Epoch 71/100
16/16 - 0s - loss: 27.1624 - val_loss: 39.9359 - 34ms/epoch - 2ms/step
Epoch 72/100
16/16 - 0s - loss: 27.2225 - val_loss: 39.8287 - 35ms/epoch - 2ms/step
Epoch 73/100
16/16 - 0s - loss: 27.2348 - val_loss: 39.8648 - 35ms/epoch - 2ms/step
Epoch 74/100
16/16 - 0s - loss: 27.1797 - val_loss: 39.9395 - 34ms/epoch - 2ms/step
Epoch 75/100
16/16 - 0s - loss: 27.2022 - val_loss: 40.0205 - 37ms/epoch - 2ms/step
Epoch 76/100
16/16 - 0s - loss: 27.1748 - val_loss: 39.8796 - 36ms/epoch - 2ms/step
Epoch 77/100
16/16 - 0s - loss: 27.2081 - val_loss: 39.7826 - 35ms/epoch - 2ms/step
Epoch 78/100
16/16 - 0s - loss: 27.2132 - val_loss: 39.8522 - 36ms/epoch - 2ms/step
Epoch

Epoch 64/100
16/16 - 0s - loss: 27.1998 - val_loss: 39.9766 - 34ms/epoch - 2ms/step
Epoch 65/100
16/16 - 0s - loss: 27.1550 - val_loss: 39.9032 - 36ms/epoch - 2ms/step
Epoch 66/100
16/16 - 0s - loss: 27.1811 - val_loss: 39.8889 - 35ms/epoch - 2ms/step
Epoch 67/100
16/16 - 0s - loss: 27.2060 - val_loss: 39.9873 - 33ms/epoch - 2ms/step
Epoch 68/100
16/16 - 0s - loss: 27.1805 - val_loss: 39.8383 - 34ms/epoch - 2ms/step
Epoch 69/100
16/16 - 0s - loss: 27.1724 - val_loss: 39.8370 - 34ms/epoch - 2ms/step
Epoch 70/100
16/16 - 0s - loss: 27.2049 - val_loss: 39.8711 - 33ms/epoch - 2ms/step
Epoch 71/100
16/16 - 0s - loss: 27.1683 - val_loss: 40.0022 - 34ms/epoch - 2ms/step
Epoch 72/100
16/16 - 0s - loss: 27.1779 - val_loss: 39.9345 - 34ms/epoch - 2ms/step
Epoch 73/100
16/16 - 0s - loss: 27.2142 - val_loss: 39.9166 - 35ms/epoch - 2ms/step
Epoch 74/100
16/16 - 0s - loss: 27.1650 - val_loss: 39.8353 - 36ms/epoch - 2ms/step
Epoch 75/100
16/16 - 0s - loss: 27.1809 - val_loss: 39.8610 - 36ms/epoch - 2

16/16 - 0s - loss: 27.1208 - val_loss: 39.8372 - 33ms/epoch - 2ms/step
Epoch 61/100
16/16 - 0s - loss: 27.1523 - val_loss: 39.9508 - 35ms/epoch - 2ms/step
Epoch 62/100
16/16 - 0s - loss: 27.1850 - val_loss: 39.9897 - 35ms/epoch - 2ms/step
Epoch 63/100
16/16 - 0s - loss: 27.1300 - val_loss: 39.8022 - 35ms/epoch - 2ms/step
Epoch 64/100
16/16 - 0s - loss: 27.1698 - val_loss: 39.8536 - 35ms/epoch - 2ms/step
Epoch 65/100
16/16 - 0s - loss: 27.1481 - val_loss: 39.9213 - 34ms/epoch - 2ms/step
Epoch 66/100
16/16 - 0s - loss: 27.1317 - val_loss: 39.9886 - 36ms/epoch - 2ms/step
Epoch 67/100
16/16 - 0s - loss: 27.1856 - val_loss: 39.8128 - 36ms/epoch - 2ms/step
Epoch 68/100
16/16 - 0s - loss: 27.1888 - val_loss: 39.7737 - 35ms/epoch - 2ms/step
Epoch 69/100
16/16 - 0s - loss: 27.1293 - val_loss: 39.9279 - 36ms/epoch - 2ms/step
Epoch 70/100
16/16 - 0s - loss: 27.1136 - val_loss: 39.9440 - 35ms/epoch - 2ms/step
Epoch 71/100
16/16 - 0s - loss: 27.1700 - val_loss: 40.0178 - 40ms/epoch - 2ms/step
Epoch

Epoch 57/100
16/16 - 0s - loss: 27.0668 - val_loss: 39.9540 - 41ms/epoch - 3ms/step
Epoch 58/100
16/16 - 0s - loss: 27.0549 - val_loss: 39.9021 - 38ms/epoch - 2ms/step
Epoch 59/100
16/16 - 0s - loss: 27.0897 - val_loss: 39.9937 - 34ms/epoch - 2ms/step
Epoch 60/100
16/16 - 0s - loss: 27.0487 - val_loss: 39.8618 - 35ms/epoch - 2ms/step
Epoch 61/100
16/16 - 0s - loss: 27.1145 - val_loss: 39.9589 - 35ms/epoch - 2ms/step
Epoch 62/100
16/16 - 0s - loss: 27.0667 - val_loss: 39.9025 - 32ms/epoch - 2ms/step
Epoch 63/100
16/16 - 0s - loss: 27.0861 - val_loss: 39.7966 - 34ms/epoch - 2ms/step
Epoch 64/100
16/16 - 0s - loss: 27.0880 - val_loss: 39.9024 - 36ms/epoch - 2ms/step
Epoch 65/100
16/16 - 0s - loss: 27.1326 - val_loss: 39.9787 - 41ms/epoch - 3ms/step
Epoch 66/100
16/16 - 0s - loss: 27.1474 - val_loss: 39.7971 - 37ms/epoch - 2ms/step
Epoch 67/100
16/16 - 0s - loss: 27.0719 - val_loss: 39.9980 - 35ms/epoch - 2ms/step
Epoch 68/100
16/16 - 0s - loss: 27.0510 - val_loss: 39.9141 - 36ms/epoch - 2

16/16 - 0s - loss: 27.0164 - val_loss: 39.9371 - 39ms/epoch - 2ms/step
Epoch 54/100
16/16 - 0s - loss: 27.0225 - val_loss: 39.8610 - 54ms/epoch - 3ms/step
Epoch 55/100
16/16 - 0s - loss: 27.0217 - val_loss: 39.8853 - 36ms/epoch - 2ms/step
Epoch 56/100
16/16 - 0s - loss: 27.0317 - val_loss: 39.8693 - 37ms/epoch - 2ms/step
Epoch 57/100
16/16 - 0s - loss: 26.9993 - val_loss: 39.9228 - 37ms/epoch - 2ms/step
Epoch 58/100
16/16 - 0s - loss: 27.0230 - val_loss: 40.0502 - 33ms/epoch - 2ms/step
Epoch 59/100
16/16 - 0s - loss: 27.0317 - val_loss: 39.8701 - 36ms/epoch - 2ms/step
Epoch 60/100
16/16 - 0s - loss: 27.0094 - val_loss: 39.9417 - 35ms/epoch - 2ms/step
Epoch 61/100
16/16 - 0s - loss: 27.0055 - val_loss: 39.9965 - 37ms/epoch - 2ms/step
Epoch 62/100
16/16 - 0s - loss: 26.9965 - val_loss: 39.8468 - 34ms/epoch - 2ms/step
Epoch 63/100
16/16 - 0s - loss: 27.0334 - val_loss: 39.8724 - 34ms/epoch - 2ms/step
Epoch 64/100
16/16 - 0s - loss: 27.0787 - val_loss: 40.0062 - 38ms/epoch - 2ms/step
Epoch

Epoch 50/100
16/16 - 0s - loss: 26.9911 - val_loss: 40.0505 - 38ms/epoch - 2ms/step
Epoch 51/100
16/16 - 0s - loss: 26.9805 - val_loss: 40.0120 - 37ms/epoch - 2ms/step
Epoch 52/100
16/16 - 0s - loss: 26.9586 - val_loss: 39.9337 - 40ms/epoch - 2ms/step
Epoch 53/100
16/16 - 0s - loss: 27.0097 - val_loss: 39.8775 - 40ms/epoch - 2ms/step
Epoch 54/100
16/16 - 0s - loss: 27.0412 - val_loss: 40.1236 - 37ms/epoch - 2ms/step
Epoch 55/100
16/16 - 0s - loss: 26.9772 - val_loss: 39.9152 - 40ms/epoch - 2ms/step
Epoch 56/100
16/16 - 0s - loss: 26.9838 - val_loss: 39.9389 - 36ms/epoch - 2ms/step
Epoch 57/100
16/16 - 0s - loss: 26.9917 - val_loss: 39.7956 - 38ms/epoch - 2ms/step
Epoch 58/100
16/16 - 0s - loss: 27.0158 - val_loss: 39.8955 - 40ms/epoch - 2ms/step
Epoch 59/100
16/16 - 0s - loss: 26.9804 - val_loss: 39.9011 - 40ms/epoch - 2ms/step
Epoch 60/100
16/16 - 0s - loss: 26.9551 - val_loss: 39.9824 - 37ms/epoch - 2ms/step
Epoch 61/100
16/16 - 0s - loss: 27.0063 - val_loss: 39.9000 - 37ms/epoch - 2

16/16 - 0s - loss: 26.9482 - val_loss: 39.9584 - 35ms/epoch - 2ms/step
Epoch 47/100
16/16 - 0s - loss: 26.9564 - val_loss: 39.9025 - 35ms/epoch - 2ms/step
Epoch 48/100
16/16 - 0s - loss: 26.9328 - val_loss: 39.9519 - 36ms/epoch - 2ms/step
Epoch 49/100
16/16 - 0s - loss: 26.9625 - val_loss: 40.0377 - 36ms/epoch - 2ms/step
Epoch 50/100
16/16 - 0s - loss: 26.9574 - val_loss: 39.8736 - 36ms/epoch - 2ms/step
Epoch 51/100
16/16 - 0s - loss: 26.9583 - val_loss: 39.8497 - 36ms/epoch - 2ms/step
Epoch 52/100
16/16 - 0s - loss: 27.0069 - val_loss: 39.8982 - 38ms/epoch - 2ms/step
Epoch 53/100
16/16 - 0s - loss: 27.0267 - val_loss: 40.0536 - 37ms/epoch - 2ms/step
Epoch 54/100
16/16 - 0s - loss: 26.9803 - val_loss: 39.8919 - 37ms/epoch - 2ms/step
Epoch 55/100
16/16 - 0s - loss: 26.9613 - val_loss: 39.9611 - 36ms/epoch - 2ms/step
Epoch 56/100
16/16 - 0s - loss: 26.9577 - val_loss: 39.8691 - 32ms/epoch - 2ms/step
Epoch 57/100
16/16 - 0s - loss: 26.9442 - val_loss: 39.9833 - 37ms/epoch - 2ms/step
Epoch

Epoch 43/100
16/16 - 0s - loss: 26.9096 - val_loss: 39.9006 - 37ms/epoch - 2ms/step
Epoch 44/100
16/16 - 0s - loss: 26.9244 - val_loss: 39.9469 - 35ms/epoch - 2ms/step
Epoch 45/100
16/16 - 0s - loss: 26.9148 - val_loss: 39.8156 - 35ms/epoch - 2ms/step
Epoch 46/100
16/16 - 0s - loss: 26.9345 - val_loss: 39.9447 - 36ms/epoch - 2ms/step
Epoch 47/100
16/16 - 0s - loss: 26.9221 - val_loss: 39.8812 - 36ms/epoch - 2ms/step
Epoch 48/100
16/16 - 0s - loss: 26.9151 - val_loss: 39.8494 - 39ms/epoch - 2ms/step
Epoch 49/100
16/16 - 0s - loss: 26.9566 - val_loss: 40.0738 - 37ms/epoch - 2ms/step
Epoch 50/100
16/16 - 0s - loss: 26.8960 - val_loss: 39.8931 - 36ms/epoch - 2ms/step
Epoch 51/100
16/16 - 0s - loss: 26.9106 - val_loss: 39.9872 - 36ms/epoch - 2ms/step
Epoch 52/100
16/16 - 0s - loss: 26.8873 - val_loss: 39.9518 - 34ms/epoch - 2ms/step
Epoch 53/100
16/16 - 0s - loss: 26.8938 - val_loss: 39.8849 - 34ms/epoch - 2ms/step
Epoch 54/100
16/16 - 0s - loss: 26.9206 - val_loss: 39.9118 - 38ms/epoch - 2

16/16 - 0s - loss: 26.9347 - val_loss: 39.8521 - 37ms/epoch - 2ms/step
Epoch 40/100
16/16 - 0s - loss: 26.9098 - val_loss: 39.8895 - 40ms/epoch - 2ms/step
Epoch 41/100
16/16 - 0s - loss: 26.9471 - val_loss: 40.0819 - 54ms/epoch - 3ms/step
Epoch 42/100
16/16 - 0s - loss: 26.8616 - val_loss: 39.8359 - 39ms/epoch - 2ms/step
Epoch 43/100
16/16 - 0s - loss: 26.9072 - val_loss: 39.8513 - 37ms/epoch - 2ms/step
Epoch 44/100
16/16 - 0s - loss: 26.8962 - val_loss: 39.8554 - 36ms/epoch - 2ms/step
Epoch 45/100
16/16 - 0s - loss: 26.8728 - val_loss: 39.8963 - 37ms/epoch - 2ms/step
Epoch 46/100
16/16 - 0s - loss: 26.8719 - val_loss: 39.7688 - 37ms/epoch - 2ms/step
Epoch 47/100
16/16 - 0s - loss: 26.8635 - val_loss: 39.7816 - 34ms/epoch - 2ms/step
Epoch 48/100
16/16 - 0s - loss: 26.8942 - val_loss: 39.9656 - 34ms/epoch - 2ms/step
Epoch 49/100
16/16 - 0s - loss: 26.9058 - val_loss: 39.7702 - 35ms/epoch - 2ms/step
Epoch 50/100
16/16 - 0s - loss: 26.8652 - val_loss: 39.9436 - 37ms/epoch - 2ms/step
Epoch

Epoch 36/100
16/16 - 0s - loss: 26.5695 - val_loss: 39.9534 - 44ms/epoch - 3ms/step
Epoch 37/100
16/16 - 0s - loss: 26.5622 - val_loss: 39.9496 - 38ms/epoch - 2ms/step
Epoch 38/100
16/16 - 0s - loss: 26.5678 - val_loss: 40.0766 - 39ms/epoch - 2ms/step
Epoch 39/100
16/16 - 0s - loss: 26.5547 - val_loss: 39.9730 - 36ms/epoch - 2ms/step
Epoch 40/100
16/16 - 0s - loss: 26.5539 - val_loss: 39.9042 - 33ms/epoch - 2ms/step
Epoch 41/100
16/16 - 0s - loss: 26.5448 - val_loss: 39.9115 - 35ms/epoch - 2ms/step
Epoch 42/100
16/16 - 0s - loss: 26.5731 - val_loss: 40.1114 - 34ms/epoch - 2ms/step
Epoch 43/100
16/16 - 0s - loss: 26.5256 - val_loss: 39.9706 - 34ms/epoch - 2ms/step
Epoch 44/100
16/16 - 0s - loss: 26.5546 - val_loss: 39.9132 - 35ms/epoch - 2ms/step
Epoch 45/100
16/16 - 0s - loss: 26.5410 - val_loss: 40.0219 - 37ms/epoch - 2ms/step
Epoch 46/100
16/16 - 0s - loss: 26.5621 - val_loss: 40.0291 - 37ms/epoch - 2ms/step
Epoch 47/100
16/16 - 0s - loss: 26.6212 - val_loss: 40.1877 - 34ms/epoch - 2

16/16 - 0s - loss: 26.3964 - val_loss: 39.9813 - 34ms/epoch - 2ms/step
Epoch 33/100
16/16 - 0s - loss: 26.4127 - val_loss: 40.0612 - 31ms/epoch - 2ms/step
Epoch 34/100
16/16 - 0s - loss: 26.4170 - val_loss: 40.2545 - 35ms/epoch - 2ms/step
Epoch 35/100
16/16 - 0s - loss: 26.3785 - val_loss: 40.0851 - 37ms/epoch - 2ms/step
Epoch 36/100
16/16 - 0s - loss: 26.4264 - val_loss: 40.0484 - 34ms/epoch - 2ms/step
Epoch 37/100
16/16 - 0s - loss: 26.3860 - val_loss: 40.1089 - 37ms/epoch - 2ms/step
Epoch 38/100
16/16 - 0s - loss: 26.3612 - val_loss: 40.1719 - 37ms/epoch - 2ms/step
Epoch 39/100
16/16 - 0s - loss: 26.3940 - val_loss: 40.0680 - 36ms/epoch - 2ms/step
Epoch 40/100
16/16 - 0s - loss: 26.3606 - val_loss: 40.1228 - 37ms/epoch - 2ms/step
Epoch 41/100
16/16 - 0s - loss: 26.4159 - val_loss: 40.2507 - 37ms/epoch - 2ms/step
Epoch 42/100
16/16 - 0s - loss: 26.4108 - val_loss: 40.0081 - 36ms/epoch - 2ms/step
Epoch 43/100
16/16 - 0s - loss: 26.3976 - val_loss: 40.1099 - 35ms/epoch - 2ms/step
Epoch

Epoch 29/100
16/16 - 0s - loss: 26.2885 - val_loss: 40.1109 - 36ms/epoch - 2ms/step
Epoch 30/100
16/16 - 0s - loss: 26.2571 - val_loss: 40.0946 - 34ms/epoch - 2ms/step
Epoch 31/100
16/16 - 0s - loss: 26.3234 - val_loss: 40.0646 - 36ms/epoch - 2ms/step
Epoch 32/100
16/16 - 0s - loss: 26.2988 - val_loss: 39.9810 - 35ms/epoch - 2ms/step
Epoch 33/100
16/16 - 0s - loss: 26.2662 - val_loss: 40.0695 - 36ms/epoch - 2ms/step
Epoch 34/100
16/16 - 0s - loss: 26.2944 - val_loss: 40.1536 - 35ms/epoch - 2ms/step
Epoch 35/100
16/16 - 0s - loss: 26.3185 - val_loss: 40.0045 - 35ms/epoch - 2ms/step
Epoch 36/100
16/16 - 0s - loss: 26.2763 - val_loss: 40.1396 - 37ms/epoch - 2ms/step
Epoch 37/100
16/16 - 0s - loss: 26.3304 - val_loss: 39.9297 - 39ms/epoch - 2ms/step
Epoch 38/100
16/16 - 0s - loss: 26.2481 - val_loss: 40.1365 - 38ms/epoch - 2ms/step
Epoch 39/100
16/16 - 0s - loss: 26.2848 - val_loss: 40.1029 - 37ms/epoch - 2ms/step
Epoch 40/100
16/16 - 0s - loss: 26.2799 - val_loss: 40.0933 - 36ms/epoch - 2

16/16 - 0s - loss: 25.9471 - val_loss: 39.4918 - 34ms/epoch - 2ms/step
Epoch 26/100
16/16 - 0s - loss: 25.9166 - val_loss: 39.4371 - 35ms/epoch - 2ms/step
Epoch 27/100
16/16 - 0s - loss: 25.9171 - val_loss: 39.5722 - 36ms/epoch - 2ms/step
Epoch 28/100
16/16 - 0s - loss: 25.9295 - val_loss: 39.5372 - 36ms/epoch - 2ms/step
Epoch 29/100
16/16 - 0s - loss: 25.9157 - val_loss: 39.5511 - 36ms/epoch - 2ms/step
Epoch 30/100
16/16 - 0s - loss: 25.9201 - val_loss: 39.3939 - 36ms/epoch - 2ms/step
Epoch 31/100
16/16 - 0s - loss: 25.9066 - val_loss: 39.4977 - 36ms/epoch - 2ms/step
Epoch 32/100
16/16 - 0s - loss: 25.8709 - val_loss: 39.4646 - 37ms/epoch - 2ms/step
Epoch 33/100
16/16 - 0s - loss: 25.9036 - val_loss: 39.4544 - 36ms/epoch - 2ms/step
Epoch 34/100
16/16 - 0s - loss: 25.9141 - val_loss: 39.5148 - 65ms/epoch - 4ms/step
Epoch 35/100
16/16 - 0s - loss: 25.9015 - val_loss: 39.3661 - 34ms/epoch - 2ms/step
Epoch 36/100
16/16 - 0s - loss: 25.8815 - val_loss: 39.3601 - 36ms/epoch - 2ms/step
Epoch

Epoch 22/100
16/16 - 0s - loss: 25.6667 - val_loss: 39.3978 - 34ms/epoch - 2ms/step
Epoch 23/100
16/16 - 0s - loss: 25.6250 - val_loss: 39.4011 - 34ms/epoch - 2ms/step
Epoch 24/100
16/16 - 0s - loss: 25.6116 - val_loss: 39.3820 - 36ms/epoch - 2ms/step
Epoch 25/100
16/16 - 0s - loss: 25.6290 - val_loss: 39.4223 - 35ms/epoch - 2ms/step
Epoch 26/100
16/16 - 0s - loss: 25.6641 - val_loss: 39.6314 - 33ms/epoch - 2ms/step
Epoch 27/100
16/16 - 0s - loss: 25.6103 - val_loss: 39.3724 - 35ms/epoch - 2ms/step
Epoch 28/100
16/16 - 0s - loss: 25.6096 - val_loss: 39.3476 - 36ms/epoch - 2ms/step
Epoch 29/100
16/16 - 0s - loss: 25.5970 - val_loss: 39.3268 - 35ms/epoch - 2ms/step
Epoch 30/100
16/16 - 0s - loss: 25.6442 - val_loss: 39.4179 - 36ms/epoch - 2ms/step
Epoch 31/100
16/16 - 0s - loss: 25.6600 - val_loss: 39.3133 - 36ms/epoch - 2ms/step
Epoch 32/100
16/16 - 0s - loss: 25.6011 - val_loss: 39.3545 - 37ms/epoch - 2ms/step
Epoch 33/100
16/16 - 0s - loss: 25.6172 - val_loss: 39.3863 - 37ms/epoch - 2

16/16 - 0s - loss: 25.5259 - val_loss: 39.6257 - 33ms/epoch - 2ms/step
Epoch 19/100
16/16 - 0s - loss: 25.4617 - val_loss: 39.4816 - 33ms/epoch - 2ms/step
Epoch 20/100
16/16 - 0s - loss: 25.5078 - val_loss: 39.1780 - 35ms/epoch - 2ms/step
Epoch 21/100
16/16 - 0s - loss: 25.4603 - val_loss: 39.3560 - 34ms/epoch - 2ms/step
Epoch 22/100
16/16 - 0s - loss: 25.4635 - val_loss: 39.3175 - 35ms/epoch - 2ms/step
Epoch 23/100
16/16 - 0s - loss: 25.5228 - val_loss: 39.5358 - 48ms/epoch - 3ms/step
Epoch 24/100
16/16 - 0s - loss: 25.4967 - val_loss: 39.3433 - 35ms/epoch - 2ms/step
Epoch 25/100
16/16 - 0s - loss: 25.4467 - val_loss: 39.2209 - 36ms/epoch - 2ms/step
Epoch 26/100
16/16 - 0s - loss: 25.4746 - val_loss: 39.3287 - 36ms/epoch - 2ms/step
Epoch 27/100
16/16 - 0s - loss: 25.4641 - val_loss: 39.3692 - 34ms/epoch - 2ms/step
Epoch 28/100
16/16 - 0s - loss: 25.4675 - val_loss: 39.4221 - 34ms/epoch - 2ms/step
Epoch 29/100
16/16 - 0s - loss: 25.4576 - val_loss: 39.4343 - 34ms/epoch - 2ms/step
Epoch

<a id="item4"></a>

<a id="item4"></a>

The mean from part B (with 50 epochs): 63.55776134625607

The mean from part C (with 100 epochs): 46.65215624259431

The mean in step C is less than in B